### PR TITLE
Move to CronJobs and YAML

### PR DESCRIPTION
**Pull request by makentenza**
#### What is this PR About?
This PR will cover the issue #12 as well as moving from JSON to YAML for these templates. I have also added more objects to the templates so no commands are required to create Service Accounts or add this to a specific role. Is now aligned with the LDAP template and more aligned to be issued with openshift-applier from CASL.

#### How do we test this?
Follow instructions from README

cc: @redhat-cop/day-in-the-life-ops

### CODE REVIEW

File: jobs/README.md [hunk 1]
```diff
@@ -1,38 +1,32 @@
...
+The [scheduledjob-prune-images.yml](scheduledjob-prune-images.yml) facilitates [image pruning](https://docs.openshift.com/container-platform/latest/admin_guide/pruning_resources.html#pruning-images) of the integrated docker registry while the [scheduledjob-prune-builds-deployments.yml](scheduledjob-prune-builds-deployments.yml) facilitates pruning [builds](https://docs.openshift.com/container-platform/latest/admin_guide/pruning_resources.html#pruning-builds) and [deployments](https://docs.openshift.com/container-platform/latest/admin_guide/pruning_resources.html#pruning-deployments) on a regular basis.
```
Comment (sabre1041) [inline jobs/README.md]: Would it be possible to rename the files from scheduledjob to cronjob to correspond with the changes in this PR?

**Review by sabre1041 — DISMISSED** [review]

File: jobs/README.md (hunk 1 above)
Comment (makentenza) [reply jobs/README.md]: Sure @sabre1041, it's done now.

**Comment by oybed** [comment]
@sabre1041 @makentenza I agree with correctly naming the file, but unfortunately this will break a lot of inventories that already references this existing file. To properly "deprecate" it, how about creating a symlink to the new file that's named after the old one - ie..: scheduledjob-ldap-group-sync.yml → cronjob-ldap-group-sync.yml - will that work for referencing it remotely with `https://raw.githubusercontent ......`??

**Comment by makentenza** [comment]
@oybed just added your suggestion. Can we track those inventories so we can remove those links once they are not used anymore?

**Comment by sabre1041** [comment]
Can we look at creating a tag and referencing it to not break apps moving forward?

**Comment by makentenza** [comment]
We could do that @sabre1041, but the result should be the same and the inventories @oybed mentioned will need to be changed as well. Also, @oybed while committing the soft links to GitHub, they can't be referenced in the "raw" url (or at least I didn't manage to get that working with soft links, just with hard links) so this will probably not address what you want

**Comment by oybed** [comment]
@makentenza ok, yes not being able to reference them with the symlinks is what I was afraid of (hence the question earlier). so it sounds to me like we don't really have a good solution for deprecating content as-is. One solution would be to cut releases (or just use tags like @sabre1041 suggested) and make consumers use these. 

@makentenza as for keeping tabs on / track the consumers - that's impossible as we don't know who all has pointed to these files.

**Comment by makentenza** [comment]
@oybed I agree that we don't have a solution for deprecating content, but do we really require releases for small repositories like this? Maybe I'm wrong but that would require an additional effort from our side and not sure if it worths for this particular. From my point of view, a consumer just uses these repos once per cluster and generally speaking people try to use the latest OpenShift release available while deploying a cluster. So as long we do an effort to maintain the repository aligned with the OpenShift release we will be fine. Kubernetes API is moving really fast, also this CrobJob will be deprecated as 'alpha' in 1.8 and moved to 'beta'.

**Comment by etsauer** [comment]
@oybed @sabre1041 @makentenza I vote we go ahead and start cutting releases.

**Comment by oybed** [comment]
@etsauer agreed. @makentenza cutting release is basically to "protect" both the consumers and us. I.e.: the consumed can continue to use it knowing that it won't change underneath them, while we can continue to evolve and change the master (and newer releases) knowing that we don't impact anybody by doing so. The change you are making is a good example of this - i.e.: it will break multiple of the consumers that I know of, and I'm sure there are others that we don't know about.

**Comment by oybed** [comment]
`v.3.6.0` release was cut - will merge this into `master`
https://github.com/redhat-cop/openshift-management/releases/tag/v3.6.0

**Review by oybed — APPROVED** [review]
All comments have been incorporated and a release has been cut for backwards features and functionality.

**Pull request merged**